### PR TITLE
Enabling Proxy CMD

### PR DESCRIPTION
**Pull request by scbedd**
This PR...

- [x] Brings the CLI improvements to our docker calls. Users can have `docker` invoke a restore or push for them. The only requirement is that they set a couple environment variables when running the image.
  - The default is _still_ `run the server` even using `CMD` instead of `ENTRYPOINT`, the difference is now the different proxy cli modes are available EG: `docker run <image> test-proxy push sdk/tables/assets.json`
- [x] Removes windows docker image.
  - Originally, the test-proxy docker image was consumed in CI. On windows agents, `docker` was running in `windows` mode without a way to swap to `linux` mode. This meant that we needed to provide a windows docker image if we wanted to support CI usage.
  - We maintained both until December 2021, at which point the docker site-license changes precluded us from using `docker` on our mac agents. This effectively killed `test-proxy-docker` usage in CI. 
  - Given that no one uses windows images locally, and we can't use in CI anyway due to mac restrictions, I am deprecating and deleting the windows docker image. It is quite burdensome to keep updated vs the relatively simple alpine linux image.

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs [hunk 1]
```diff
@@ -89,7 +89,7 @@ public static void LogDebug(string details)
             }
             else
             {
-                Debug.WriteLine(details);
+                System.Console.WriteLine(details);
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs:92]: Discovered that `debug.writeline` doesn't show up by default on command line calling the `test-proxy` tool. Given that, adjusted to console out directly.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 2]
```diff
@@ -95,13 +95,13 @@ If you've already installed the tool, you can always check the installed version
 The Azure SDK Team maintains a public Azure Container Registry.
 
 ```powershell
-> docker run -v <your-volume-name-or-location>:/srv/testproxy/ -p 5001:5001 -p 5000:5000 azsdkengsys.azurecr.io/engsys/testproxy-lin:latest
+> docker run -v <your-volume-name-or-location>:/srv/testproxy/ -p 5001:5001 -p 5000:5000 azsdkengsys.azurecr.io/engsys/test-proxy:latest
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:98]: This is merely formalizing the multiplatform image `engsys/test-proxy`

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 3]
```diff
@@ -115,29 +115,38 @@ The azure-sdk team regularly update the image associated with the `latest` tag.
...
 ```
 
 ## Command line arguments
 
 This is the help information for test-proxy. It uses the nuget package [`CommandLineParser`](https://www.nuget.org/packages/CommandLineParser) to parse arguments.
 
+The test-proxy executable fulfills one of two primary purposes:
+
+1. The test-proxy server (the only option up to this point)
+2. `asset-sync` push/restore/reset.
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:128]:
```suggestion
2. [`asset-sync`](#asset-sync-retrieve-external-test-recordings) push/restore/reset.
```

Comment (benbp) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:128]: If you take this, the backticks might need to surround the whole thing? I forget how markdown resolves links + code blocks together.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 4]
```diff
@@ -115,29 +115,38 @@ The azure-sdk team regularly update the image associated with the `latest` tag.
...
+c Microsoft Corporation. All rights reserved.
+
+  start      (Default Verb) Start the TestProxy.
+
+  push       Push the assets, referenced by assets.json, into git.
 
-Usage:
-  Azure.Sdk.Tools.TestProxy [options] [<args>...]
+  reset      Reset the assets, referenced by assets.json, from git to their original files referenced by the tag. Will prompt
+             if there's pending changes.
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:143]: I deserve to burn for this level of grammar nitpicking:

```suggestion
             if there are pending changes.
```

**Review by benbp — APPROVED** [review]

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md (hunk 4 above)
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:143]: Technically correct is the best type of correct.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md (hunk 3 above)
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:128]: Tested locally and it works just fine. Takin it.

**Review by JimSuplizio — APPROVED** [review]

**Pull request merged**